tasks.c

```diff
@@ -7653,3 +7653,52 @@ static void prvAddCurrentTaskToDelayedList( TickType_t xTicksToWait,
     #endif
 
 #endif /* if ( configINCLUDE_FREERTOS_TASK_C_ADDITIONS_H == 1 ) */
+/*-----------------------------------------------------------*/
+
+#if ( ( configSUPPORT_STATIC_ALLOCATION == 1 ) && ( configKERNEL_PROVIDED_STATIC_MEMORY == 1 ) )
+
+/*
+ * This is the kernel provided implementation of vApplicationGetIdleTaskMemory()
+ * to provide the memory that is used by the Idle task. It is used when
+ * configKERNEL_PROVIDED_STATIC_MEMORY is set to 1. The application can provide
+ * it's own implementation of vApplicationGetIdleTaskMemory by setting
+ * configKERNEL_PROVIDED_STATIC_MEMORY to 0 or leaving it undefined.
+ */
+    void vApplicationGetIdleTaskMemory( StaticTask_t ** ppxIdleTaskTCBBuffer,
+                                        StackType_t ** ppxIdleTaskStackBuffer,
+                                        uint32_t * pulIdleTaskStackSize )
+    {
+        static StaticTask_t xIdleTaskTCB;
```

Comment:
Should these receive the [PRIVILEGED_DATA](https://github.com/FreeRTOS/FreeRTOS-Kernel/blob/V10.6.1/include/mpu_wrappers.h#L240) flag?
With this implementation If people end up using this weak implementation the TCB for this struct would live in SRAM outside of that region. Where then any unprivileged task could modify the values inside of the TCB (Such as ulContext). Which could potentially be used to gain access to privileged APIs and data from an unprivileged task.

Reply:
Good catch. Done.

> Where then any unprivileged task could modify the values inside of the TCB (Such as ulContext). Which could potentially be used to gain access to privileged APIs and data from an unprivileged task.

Even without tagging this as privileged data, that is not true. An unprivileged task will not have access to this data unless explicitly granted to.